heron/tmaster/src/cpp/manager/stats-interface.cpp

```diff
@@ -137,6 +143,33 @@ void StatsInterface::HandleExceptionSummaryRequest(IncomingHTTPRequest* _request
   LOG(INFO) << "Returned exceptions response";
 }
 
+void StatsInterface::HandleStmgrsRegistrationSummaryRequest(IncomingHTTPRequest* _request) {
+  LOG(INFO) << "Request for stream managers registration summary " << _request->GetQuery();
+  unsigned char* request_data =
+    _request->ExtractFromPostData(0, _request->GetPayloadSize());
+  heron::proto::tmaster::StmgrsRegistrationSummaryRequest stmgrs_reg_request;
+  if (!stmgrs_reg_request.ParseFromArray(request_data, _request->GetPayloadSize())) {
+    LOG(ERROR) << "Unable to deserialize post data specified in" <<
+      "StmgrsRegistrationSummaryRequest" << std::endl;
+    http_server_->SendErrorReply(_request, 400);
+    delete _request;
+    return;
+  }
+  auto stmgrs_reg_summary_response = tmaster_->GetStmgrsRegSummary();
+  sp_string response_string;
+  CHECK(stmgrs_reg_summary_response->SerializeToString(&response_string));
+  OutgoingHTTPResponse* http_response = new OutgoingHTTPResponse(_request);
+  http_response->AddHeader("Content-Type", "application/octet-stream");
+  std::ostringstream length_str;
+  length_str << response_string.size();
+  http_response->AddHeader("Content-Length", length_str.str());
```

Comment:
`std::string(length_str)`? Fails to compile using Clang (Apple LLVM version 8.0.0 (clang-800.0.38)).

Error message:
```
ERROR: .../heron/tmaster/src/cpp/BUILD:3:1: C++ compilation of rule '//heron/tmaster/src/cpp:tmaster-cxx' failed: cc_wrapper.sh failed: error executing command external/local_config_cc/cc_wrapper.sh -U_FORTIFY_SOURCE '-D_FORTIFY_SOURCE=1' -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer '-std=c++0x' -MD -MF ... (remaining 60 argument(s) skipped): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
heron/tmaster/src/cpp/manager/stats-interface.cpp:165:46: error: no matching conversion for functional-style cast from 'std::ostringstream' (aka 'basic_ostringstream<char>') to 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char> >')
  http_response->AddHeader("Content-Length", std::string(length_str));
```